tests/run.go

```diff
@@ -371,38 +381,186 @@ func Run(o *option.Option) {
 				// create a file in tmpfs mount and verify it doesn't exist after stopping and restarting it
-				command.Run(o, "exec", testContainerName, "sh", "-c", fmt.Sprintf("echo foo > %s/bar.txt", tmpfsDir))
-				command.Run(o, "kill", testContainerName) // have to use kill to stop the container running with sleep infinity
-				command.Run(o, "start", testContainerName)
-				command.RunWithoutSuccessfulExit(o, "exec", testContainerName, "sh", "-c", fmt.Sprintf("cat %s/bar.txt", tmpfsDir))
+				command.Run(o.BaseOpt, "exec", testContainerName, "sh", "-c", fmt.Sprintf("echo foo > %s/bar.txt", tmpfsDir))
+				command.Run(o.BaseOpt, "kill", testContainerName) // have to use kill to stop the container running with sleep infinity
+				command.Run(o.BaseOpt, "start", testContainerName)
+				command.RunWithoutSuccessfulExit(o.BaseOpt, "exec", testContainerName, "sh", "-c", fmt.Sprintf("cat %s/bar.txt", tmpfsDir))
 			})
 
 			ginkgo.It("should mount a volume using --mount type=volume flag", func() {
-				command.Run(o, "run", "--name", testContainerName, "--mount",
+				command.Run(o.BaseOpt, "run", "--name", testContainerName, "--mount",
 					fmt.Sprintf("type=volume,source=%s,target=%s", testVolumeName, destDir), defaultImage)
-				srcDir := command.StdoutStr(o, "volume", "inspect", testVolumeName, "--format", "{{.Mountpoint}}")
+				srcDir := command.StdoutStr(o.BaseOpt, "volume", "inspect", testVolumeName, "--format", "{{.Mountpoint}}")
 				expectedMount := []MountJSON{makeMount(volumeType, srcDir, destDir, "", true)}
-				actualMount := getContainerMounts(o, testContainerName)
+				actualMount := getContainerMounts(o.BaseOpt, testContainerName)
 				verifyMountsInfo(actualMount, expectedMount)
 			})
 		})
+
+		// Cgroup version v2
+		ginkgo.When("running a container with resource flags", func() {
+			ginkgo.BeforeEach(func() {
+				if o.CGMode != Unified {
+					ginkgo.Skip("requires cgroup v2 to test resource flags")
+				}
+			})
+			ginkgo.It("should set number of CPUs with --cpus flag", func() {
+				cpuMax := command.StdoutStr(o.BaseOpt, "run", "--cpus", "0.42", "-w", "/sys/fs/cgroup", defaultImage, "cat", "cpu.max")
+				gomega.Expect(cpuMax).To(gomega.Equal("42000 100000"))
+			})
+
+			ginkgo.It("should limit CPU CFS (Completely Fair Scheduler) quota and period with --cpu-quota and --cpu-period flags", func() {
+				cpuMax := command.StdoutStr(o.BaseOpt, "run", "--cpu-quota", "42000",
+					"--cpu-period", "100000", "-w", "/sys/fs/cgroup", defaultImage, "cat", "cpu.max")
+				gomega.Expect(cpuMax).To(gomega.Equal("42000 100000"))
+			})
+
+			ginkgo.It("should set the CPU shares with --cpu-shares flag", func() {
+				// CgroupV2 CPUShares => weight := 1 + ((shares-2)*9999)/262142
+				//nolint: lll // Ref. https://github.com/google/cadvisor/blob/ce07bb28eadc18183df15ca5346293af6b020b33/integration/tests/api/docker_test.go#L216-L222
+				// Ref. https://github.com/google/cadvisor/blob/master/integration/tests/api/docker_test.go#L216-L222
```

Comment:
- The part after `//nolint: lll // ` should be the explanation.
- We may want to remove the `blob/master` link.

```suggestion
				//nolint: lll // Writing the URL in one line enables folks to directly click it.
				// Ref. https://github.com/google/cadvisor/blob/ce07bb28eadc18183df15ca5346293af6b020b33/integration/tests/api/docker_test.go#L216-L222
```